src/averageAges.js

```diff
@@ -1,5 +1,42 @@
 'use strict';
 
+const getByKeyValue = (key, value) => people => {
+  const peopleCopy = [ ...people ];
+
+  return peopleCopy.filter(person => person[key] === value);
+};
+
+const getMen = getByKeyValue('sex', 'm');
+const getWomen = getByKeyValue('sex', 'f');
+
+const getValuesArr = (key) => people => {
+  const resultArr = [];
+
+  people.filter(person => person[key] !== null
+    ? resultArr.push(person[key]) : null);
+
+  return resultArr;
+};
+
+const getMotherNames = getValuesArr('mother');
+
+const getMothers = (people, group = people) => {
+  const women = getWomen(people);
+  const motherNames = getMotherNames(group);
+  const mothers = women.filter(woman => motherNames.includes(woman['name']));
+
+  return mothers;
+};
```

Comment:
All of this to select all mothers within people? I bet you can replace this with far less code lines.

1. Pick up all women like you did `const women = getWomen(people);` ✔️
2. Filter all women who has **AT LEAST** one child (think what the function to use)❌

```suggestion
const getMothers = (people, group = people) => {
  const women = getWomen(people);
  // filter women and get mothers
  
  return mothers;
};
```

Reply:
shortened a bit
thanks